sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp

```diff
@@ -209,6 +250,8 @@ namespace Azure { namespace Core { namespace Http {
      */
     virtual void OnUpgradedConnection(std::unique_ptr<CurlNetworkConnection>&&){};
 
+    void ValidateKeepAliveHeaders(Request& request, std::unique_ptr<RawResponse>& response);
```

Comment:
Could you add some text explaining why we need to keep the request and response headers consistent? 

And more importantly, what we are expected to do if they are inconsistent, given that the server accepted our request and has sent a response.

Reply:
i did explain in the comment on the method lower down, if the headers are ot there in the response i remove them from the request and and cleanup the settings so they are not considered by the cache when the request is released back to the cache